reana_server/scheduler.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2018 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA Server Workflow Execution Scheduler."""
+
+import json
+
+from reana_commons.config import MQ_DEFAULT_QUEUES
+from reana_commons.consumer import BaseConsumer
+from reana_commons.tasks import reana_ready
+from reana_db.database import Session
+from reana_db.models import Workflow, WorkflowStatus
+
+from reana_server.api_client import current_rwc_api_client
+
+
+class WorkflowExecutionScheduler(BaseConsumer):
+    """Scheduler of workflow execution.
+
+    Class responsible for consuming from the workflow-submission queue
+    and scheduling workflows for execution based on policies and system
+    availability.
+    """
+
+    def __init__(self, **kwargs):
+        """Constructor."""
+        super(WorkflowExecutionScheduler, self).__init__(
+            queue='workflow-submission', **kwargs)
+
+    def get_consumers(self, Consumer, channel):
+        """Implement providing kombu.Consumers with queues/callbacks."""
+        return [Consumer(queues=self.queue, callbacks=[self.on_message],
+                         accept=[self.message_default_format])]
+
+    def on_message(self, workflow_submission, message):
+        """On new workflow_submission event handler."""
+        if reana_ready():
+            message.ack()
+            workflow_submission = json.loads(workflow_submission)
+            print(workflow_submission)
```

Comment:
Shall we log here or it is just a forgotten print?

Reply:
Both!